spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/source/TestDataSourceOptions.java

```diff
@@ -285,29 +286,37 @@ public void testIncrementalScanOptions() throws IOException {
             "Cannot set only end-snapshot-id for incremental scans. Please, set start-snapshot-id too.");
 
     // test (1st snapshot, current snapshot] incremental scan.
-    List<SimpleRecord> result =
+    Dataset<Row> resultDf1 =
         spark
             .read()
             .format("iceberg")
             .option("start-snapshot-id", snapshotIds.get(3).toString())
-            .load(tableLocation)
-            .orderBy("id")
-            .as(Encoders.bean(SimpleRecord.class))
-            .collectAsList();
-    Assert.assertEquals("Records should match", expectedRecords.subList(1, 4), result);
+            .load(tableLocation);
+    List<SimpleRecord> result1 =
+        resultDf1.orderBy("id").as(Encoders.bean(SimpleRecord.class)).collectAsList();
+    Assert.assertEquals("Records should match", expectedRecords.subList(1, 4), result1);
+    Assert.assertEquals("Unprocessed count should match record count", 3, resultDf1.count());
+
+    Row row1 = resultDf1.agg(functions.min("id"), functions.max("id")).head();
+    Assert.assertEquals("min value should match", 2, row1.getInt(0));
+    Assert.assertEquals("max value should match", 4, row1.getInt(1));
 
     // test (2nd snapshot, 3rd snapshot] incremental scan.
-    Dataset<Row> resultDf =
+    Dataset<Row> resultDf2 =
```

Comment:
incrementalResult (again could just be two test cases, not picky about that or naming here but we should choose one)